### DISCUSSION TITLE
unexpected merge7z.dll invocation

### DISCUSSION THREAD

**Discussion by asapuntz** [WinMerge/winmerge#601] [General]
Thx for restarting support of WinMerge!

winmergeu.exe seems to use merge7z.dll even when there's no compressed archives being manipulated

"C:\Program Files\WinMerge\WinMergeU.exe" /r /f *.h;*.c;*.cpp /wl /wr Software Software_A

GUI shows list of files are different. I double click any of them (cpp or h), and CarbonBlack (bit9) popup flags merge7z.dll

The file diff shows up fine, and subsequent file diffs don't produce additional popups, although rerunning winmerge will.

Would be nice to avoid this unexpected behavior?

**Comment by sdottaka** [comment]
I think Merge7z.dll was used to determine whether a regular file is also an archive file. It may be possible to prevent Merge7z.dll from being called by implementing a new extension-based archive file determination.

For now, to prevent calling Merge7z.dll, disable "Enable archive file support".

![image](https://user-images.githubusercontent.com/98126/106677498-290fd700-65fc-11eb-934a-514c8aeb5690.png)
